rum-injection-installer/installer/nginx_configurator.go

```diff
@@ -0,0 +1,272 @@
+// 1. Checks if a line containing the module already exists
+// 2. Creates a backup of the config file
+// 3. Appends the load_module line to the beginning of the file
+// 4. Inserts the RUM config as early as possible in the http section
+func (n *NginxConfigurator) ModifyConfig(appID, site, clientToken, agentUrl string, sessionSampleRate, sessionReplaySampleRate int) error {
+	configPath, err := n.getConfigLocation()
+	if err != nil {
+		return fmt.Errorf("failed to get config location: %v", err)
+	}
+
+	content, err := os.ReadFile(configPath)
+	if err != nil {
+		return fmt.Errorf("failed to read config file: %v", err)
+	}
+
+	// Check if Datadog module is already loaded
+	if strings.Contains(string(content), "ngx_http_datadog_module.so") {
+		return fmt.Errorf("datadog module is already loaded in the configuration")
+	}
+
+	// Create backup
+	backupPath := configPath + ".backup"
+	if err := os.WriteFile(backupPath, content, 0644); err != nil {
+		return fmt.Errorf("failed to create backup file: %v", err)
+	}
+	fmt.Printf("Backup created: %s\n", backupPath)
+
+	var newContent strings.Builder
+	newContent.WriteString(fmt.Sprintf("load_module %s/ngx_http_datadog_module.so;\n", n.ModulesPath))
+
+	// Find http and bracket { on possibly multiple lines
+	httpRegex := regexp.MustCompile(`(?m)^[\s]*http[\s]*\{`)
+
+	httpBlockStart := httpRegex.FindIndex(content)
+	if httpBlockStart == nil {
+		return fmt.Errorf("could not find http block in nginx config")
+	}
+
+	newContent.Write(content[:httpBlockStart[0]])
+
+	newContent.Write(content[httpBlockStart[0]:httpBlockStart[1]])
+	newContent.WriteString("\n")
+
+	datadogConfig := fmt.Sprintf(`    datadog_agent_url %s;
+    datadog_disable;
+    datadog_rum on;
+    datadog_rum_config "v5" {
+        "applicationId" "%s";
+        "clientToken" "%s";
+        "site" "%s";
+        "sessionSampleRate" %d;
+        "sessionReplaySampleRate" %d;
+    }
+`, agentUrl, appID, clientToken, site, sessionSampleRate, sessionReplaySampleRate)
+	newContent.WriteString(datadogConfig)
+
+	newContent.Write(content[httpBlockStart[1]:])
+
+	// Write the modified content back to the file
+	if err := os.WriteFile(configPath, []byte(newContent.String()), 0644); err != nil {
+		return fmt.Errorf("failed to write modified config: %v", err)
+	}
+
+	fmt.Printf("Successfully modified NGINX configuration file: %s\n", configPath)
+	return nil
+}
+
+// Runs 'nginx -t' to validate the configuration
+func (n *NginxConfigurator) ValidateConfig() error {
+	cmd := exec.Command("nginx", "-t")
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("nginx configuration validation failed with output:\n%s", output)
+	}
+	return nil
+}
+
+// Downloads the latest nginx-datadog for the specified architecture
+// Extracts the .so file and places it in the modules path
+func (n *NginxConfigurator) DownloadAndInstallModule(arch string) error {
+	if err := n.getModulesPath(); err != nil {
+		return err
+	}
```

Comment:
It's not strictly required to place the module in the default module path. For instance, you can store it in `/datadog/ngx_http_datadog_module.so`, and as long as the `load_module` directive specifies the absolute path, it should function correctly.

Reply:
Oh I didn't know that thanks, this was the part of the code I trusted the least, so it makes my life easier